### DISCUSSION TITLE
Webpack polyfills for Node Core Modules

### DISCUSSION THREAD

**Discussion by unonweb** [payloadcms/payload#910] [Q&A]
Hi,

I'm gettting these errors from Webpack:

![Screenshot at 2022-08-10 12-14-30](https://user-images.githubusercontent.com/79059627/183877230-36c2d78a-5184-406a-9c09-fc76e23f62f6.png)

After some research I started using these config lines for Webpack in order to reference an empty mock-object instead of the module.

```js
export default buildConfig({
	serverURL: 'http://localhost:3000',
	admin: {
		user: users.slug,
		webpack: (config) => ({
			...config, resolve: { 
				...config.resolve, alias: {
					...config.resolve.alias, fs: mockModulePath,
				}
			}
		})
	},
/*...*/
```
I must admit that I don't really understand why I'm doing this. But it worked for me.

Since today I get the following errors:

![Screenshot at 2022-08-10 12-26-02](https://user-images.githubusercontent.com/79059627/183879396-4fa8a64c-a728-4540-b508-40c8e59f71da.png)

Now however it doesn't work to use mock paths for these modules.
What do I have to do? How and where do I need to install the required polyfills?
Thanks for any support.

**Comment by unonweb** [comment]
Alright... I've solved the issue. The problem was that I was referencing in my collection some variables that were not available any more. No idea why this didn't throw a simple reference error pointing at the collection.

Furthermore I'm configuring Webpack this way now:

```js
webpack: (config) => ({
	...config, resolve: {
		...config.resolve, fallback: {
			...config.resolve.fallback,
			fs: false,
		}
	}
})
```
I don't know if this is better than the way I did before. But it works.

**Reply by leesei** [reply]
This is how I solve the issue of adding server side code to collection.

I recommend adding `resolve: { fallback: { [module]: false } }` section here:
https://payloadcms.com/docs/admin/webpack#aliasing-server-only-modules
I seems simplier than what's suggested in the doc.
